pygmt/src/coast.py

```diff
@@ -159,6 +159,11 @@ def coast(self, **kwargs):
         lake-in-island-in-lake shore. Pass a list of *level*\ /*pen*
         strings to ``shorelines`` to set multiple levels. When specific
         level pens are set, those not listed will not be drawn.
+    clip : str
+        To clip land do ``clip="land"``, ``clip="water"`` clips water. Use
+        ``clip="end"`` to mark end of existing clip path. No projection
+        information is needed. Also supply ``xshift`` and ``yshift`` settings
+        if you have moved since the clip started.
```

Comment:
> clip' actually refers to clipping of the next `grdimage` call.

That's not true. Everything, including symbols, texts, images, contours are clipped. So the clipping affects many plotting modules.

Reply:
Added an inline example.

Reply:
> > clip' actually refers to clipping of the next `grdimage` call.
> 
> That's not true. Everything, including symbols, texts, images, contours, are clipped. So the clipping affects many plotting modules.

Oh really? Then why are we making a unit test that uses `fig.grdimage` and `load_earth_age` instead of a simple `fig.plot`...

Reply:
Just to be clear, I think the inline example is good. But maybe do 2 more things:

1. Reword the docstring to explicitly mention that 'clip' is for clipping the next `grdimage`/`plot`/`contour`/etc call
2. Refactor the unit test to just use `fig.plot` or something instead of `fig.grdimage`, so that `load_earth_age` won't be needed (really sorry for the back and forth @michaelgrund!)

Reply:
> Just to be clear, I think the inline example is good. But maybe do 2 more things:
> 
>     1. Reword the docstring to explicitly mention that 'clip' is for clipping the next `grdimage`/`plot`/`contour`/etc call

Done! Used parts of @meghanrjones'  answer.

>     2. Refactor the unit test to just use `fig.plot` or something instead of `fig.grdimage`, so that `load_earth_age` won't be needed (really sorry for the back and forth @michaelgrund!)

Done! Now, simply plotted circles are used without the need of any dataset 😉